vscode-extensions: clean up

Garmelon opened the pull request:
###### Description of changes

A continuation of #223308

- Removes extensions that have vanished from the marketplace or been deprecated
- Renames an extension that moved
- Some more cleanup for consistency

During the creation of this PR, I noticed a few inconsistently capitalized extensions. I could attempt to fix them in this PR, a new PR, or just leave them be:
- `Arjun.swagger-viewer` (-> `arjun.swagger-viewer`)
- `ms-vscode.PowerShell` (-> `ms-vscode.powershell`)
- `rioj7.commandOnAllFiles` (-> `rioj7.commandonallfiles`)
- `WakaTime.vscode-wakatime` (-> `wakatime.vscode-wakatime`)

I also noticed that some aliases throw while others just point to the correct name. Not sure why.

I also noticed that `elixir-lsp.vscode-elixir-ls` is actually `jakebecker.elixir-ls` (which is currently an alias), not the other way around. I assume there is a good reason for this, as per the comment at the top of the file.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Garmelon committed:
vscode-extensions.github: split up for consistency

Garmelon committed:
vscode-extensions.coenraads.bracket-pair-colorizer: remove

The extension no longer exists on the marketplace.

Garmelon committed:
vscode-extensions.coenraads.bracket-pair-colorizer-2: remove

The extension is deprecated as its functionality is now a native feature
of VSCode. See also:
https://marketplace.visualstudio.com/items?itemName=CoenraadS.bracket-pair-colorizer-2

Garmelon committed:
vscode-extensions.faustinoaq.lex-flex-yacc-bison: remove

The extension no longer exists on the marketplace.

Garmelon committed:
vscode-extensions.msjsdiag.debugger-for-chrome: remove

The extension is deprecated as VSCode now has its own debugger that
covers the same functionality. See also:
https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome

Garmelon committed:
vscode-extensions.silvenon.mdx: remove

The extension is deprecated in favor of the official MDX extension at
https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx
See also:
https://marketplace.visualstudio.com/items?itemName=silvenon.mdx

Garmelon committed:
vscode-extensions.jpoinssonnier.vscode-styled-components: move to styled-components.vscode-styled-components, 1.4.1 -> 1.7.6

According to the readme, the old location will no longer receive
updates: https://github.com/styled-components/vscode-styled-components

Garmelon committed:
vscode-extensions.matklad.rust-analyzer: move to alias section

Garmelon committed:
vscode-extensions: restore alphabetical order

superherointj commented:
These are correct:
- Arjun.swagger-viewer
- ms-vscode.PowerShell
- rioj7.commandOnAllFiles
- WakaTime.vscode-wakatime
Notice: Upstream provides names in uppercase.

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -2010,55 +2058,7 @@ let
-          license = lib.licenses.mit;
-        };
-      };
+      ms-vscode-remote.remote-ssh = callPackage ./ms-vscode-remote.remote-ssh { };
```
`ms-vscode-remote` comes before `ms-vscode.theme`

Garmelon replied:
`ms-vscode` comes before `ms-vscode-remote` when sorting alphabetically because it is shorter.

Garmelon commented:
> As upstream provides names in uppercase, these are correct:
> * Arjun.swagger-viewer
> * ms-vscode.PowerShell
> * rioj7.commandOnAllFiles
> * WakaTime.vscode-wakatime

In that case, these are inconsistent instead:
- `angular.ng-template` -> `Angular.ng-template`
- `antyos.openscad` -> `Antyos.openscad`
- `azdavis.millet` -> `azdavis.Millet`
- `b4dm4n.vscode-nixpkgs-fmt` -> `B4dM4n.nixpkgs-fmt` (the name is different from source as well)
- `bbenoist.nix` -> `bbenoist.Nix`
- `davidanson.vscode-markdownlint` -> `DavidAnson.vscode-markdownlint`
- `divyanshuagrawal.competitive-programming-helper` -> `DivyanshuAgrawal.competitive-programming-helper`
- `editorconfig.editorconfig` -> `EditorConfig.EditorConfig`
- `elmtooling.elm-ls-vscode` -> `Elmtooling.elm-ls-vscode`
- `golang.go` -> `golang.Go`
- `graphql.vscode-graphql` -> `GraphQL.vscode-graphql`
- `gruntfuggly.todo-tree` -> `Gruntfuggly.todo-tree`
- `hookyqr.beautify` -> `HookyQR.beautify`
- `ibm.output-colorizer` -> `IBM.output-colorizer`
- `ionide.ionide-fsharp` -> `Ionide.Ionide-fsharp`
- `james-yu.latex-workshop` -> `James-Yu.latex-workshop`
and so on

superherointj commented:
> > As upstream provides names in uppercase, these are correct:
> > 
> > * Arjun.swagger-viewer
> > * ms-vscode.PowerShell
> > * rioj7.commandOnAllFiles
> > * WakaTime.vscode-wakatime
> 
> In that case, these are inconsistent instead:
> 
>     * `angular.ng-template` -> `Angular.ng-template`
> 
>     * `antyos.openscad` -> `Antyos.openscad`
> 
>     * `azdavis.millet` -> `azdavis.Millet`
> 
>     * `b4dm4n.vscode-nixpkgs-fmt` -> `B4dM4n.nixpkgs-fmt` (the name is different from source as well)
> 
>     * `bbenoist.nix` -> `bbenoist.Nix`
> 
>     * `davidanson.vscode-markdownlint` -> `DavidAnson.vscode-markdownlint`
> 
>     * `divyanshuagrawal.competitive-programming-helper` -> `DivyanshuAgrawal.competitive-programming-helper`
> 
>     * `editorconfig.editorconfig` -> `EditorConfig.EditorConfig`
> 
>     * `elmtooling.elm-ls-vscode` -> `Elmtooling.elm-ls-vscode`
> 
>     * `golang.go` -> `golang.Go`
> 
>     * `graphql.vscode-graphql` -> `GraphQL.vscode-graphql`
> 
>     * `gruntfuggly.todo-tree` -> `Gruntfuggly.todo-tree`
> 
>     * `hookyqr.beautify` -> `HookyQR.beautify`
> 
>     * `ibm.output-colorizer` -> `IBM.output-colorizer`
> 
>     * `ionide.ionide-fsharp` -> `Ionide.Ionide-fsharp`
> 
>     * `james-yu.latex-workshop` -> `James-Yu.latex-workshop`
>       and so on

:-(

What should we do about this?

Garmelon commented:
I suggest turning the four I mentioned into lowercase, adding aliases for their uppercase versions, and documenting this de-facto standard in the comment at the top of the file.

superherointj commented:
> I suggest turning the four I mentioned into lowercase, adding aliases for their uppercase versions, and documenting this de-facto standard in the comment at the top of the file.

Suggestion accepted.

Garmelon committed:
vscode-extensions: lowercase remaining extension names

In practice, the vast majority of extensions with uppercase characters
in their extension identifiers was already using lowercased versions of
the identifiers as names. This commit documents this practice and
converts the four remaining extensions.

Garmelon committed:
vscode-extensions: simplify how aliases are specified

Garmelon commented:
Apparently #222171 added the throwing aliases just a week ago. Maybe they should be turned into proper aliases instead of errors?

superherointj commented:
Jeez. I should have squashed this. But won't revert.

SuperSandro2000 commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -634,23 +634,6 @@ let
         };
       };
 
-      coenraads.bracket-pair-colorizer = buildVscodeMarketplaceExtension {
```
This is missing an alias

SuperSandro2000 commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -634,18 +634,6 @@ let
         };
       };
 
-      coenraads.bracket-pair-colorizer-2 = buildVscodeMarketplaceExtension {
```
this is also missing an alias

superherointj replied:
Sorry Sandro. I'm no longer active maintainer here.

SuperSandro2000 commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -1051,23 +1051,6 @@ let
         };
       };
 
-      faustinoaq.lex-flex-yacc-bison = buildVscodeMarketplaceExtension {
```
same

SuperSandro2000 commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -2076,18 +2076,6 @@ let
 
       ms-vsliveshare.vsliveshare = callPackage ./ms-vsliveshare.vsliveshare { };
 
-      msjsdiag.debugger-for-chrome = buildVscodeMarketplaceExtension {
```
and this

SuperSandro2000 commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -2486,18 +2486,6 @@ let
         };
       };
 
-      silvenon.mdx = buildVscodeMarketplaceExtension {
```
and here

SuperSandro2000 reviewed (commented):
You are missing throw aliases for removed extensions
